### DISCUSSION TITLE
pocketbase sidecart

### DISCUSSION THREAD

**Discussion by TobiasNickel** [pocketbase/pocketbase#5016] [Show and tell]
Hi, I build something that I name [pocketbaseSidecart](https://github.com/TobiasNickel/PocketbaseSidecart). 

This is a second binary that run next to pocketbase. it is based on bun (because `bun build --compile` is awesome). 
features:
 1. edit public files through a collection. this way you can upload a file or edit the content right in a collection.
 2. edit hooks from pb_hooks in a hooks_files collection. this way you can code some hooks and APIs without ssh or FTP into the server
 3. functions: implement functions that run in the context of `bun`. This allow using all NPM modules.
 4. CRON using those functions. 
 
 future/planned features: 
 - uploading complete archive files to the admin UI and this way deploy a new frontend app.
 - tasks: one time execution for functions in bun context.
 - reading emails from IMAP email servers
 - sending notifications such as Web-Push or mobile. (not sure if the availability of packages is enough)
 - sync features to other APIs. including other pocketbases.
 - by implementing hooks, enable Pocketbase to be Oauth Provider for other Pocketbases. (maybe that does not need a sidecart at all, I will see)
   
I think with such a [sidecart](https://github.com/TobiasNickel/PocketbaseSidecart) and those extra features. pocketbase gets interesting for more projects. 

How do you like it? What is your opinion?

**Comment by thangngoc89** [comment]
It looks interesting but beware that Pocketbase is a moving target that hasn't reached 1.0 yet and there is a big refactoring coming.
